Added file-io for state.json on client.py and data.js

neiroc-02 opened the pull request:
I ran tests on all the functions and they work. Let me know if something is inconsistent or if extra functionality is needed.

neiroc-02 committed:
State file management complete for JS side

neiroc-02 committed:
Python file io done

mrElnekave commented on Backend/data.js:
```diff
@@ -1,12 +1,157 @@
+  write_output_to_state,
+  read_pid_to_state_map,
+  write_pid_to_state_map
 };
```
Looks good, instead of read and write to PID state map. We just need the end functions, is player x alive, and kill player x. PID is read and written with the state, you don't need to have separate functions for it.

neiroc-02 replied:
Just edited the JS side and tested it, it should be fine I think.

mrElnekave commented on mpc/PythonCilent/client.py:
```diff
@@ -99,14 +100,55 @@ async def get_result_from_mpc_protocol(mpc_program):
 def populate_input_from_state():
```
Similar to the JS side, you should simply make a structure which is the state, and we read and write out for that. Having this function return this inputs_mapping isn't very helpful. Make sure that whatever state you have is used in the rest of the file as inputs to the mpc.

neiroc-02 replied:
Looks like Kenny has it covered

mrElnekave reviewed (commented):
Good first step, I think you really need to also incorporate the use of state. I also don't know if you tested this yet, but in theory this looks good.

neiroc-02 committed:
Edited data.js to remove pid_map functions and add kill player/check if player is alive functions

neiroc-02 committed:
Changed alive/dead to true/false to match python mapping better

neiroc-02 committed:
Removing python edits, just has the old stubs now

neiroc-02 committed:
removing one more thing

neiroc-02 committed:
Set backend stubs, getting ready to merge with mpc branch

neiroc-02 committed:
Removing client.py to resolve conflict

neiroc-02 committed:
Merge pull request #5 from mrElnekave/mpc-merge

Mpc merge

neiroc-02 committed:
Comments I have before development

neiroc-02 committed:
Some changes, some problems to think about on how we handle global state with individual JSONs

neiroc-02 committed:
Some phase function made and migrated to the frontend. A few phases still left to clean up.

neiroc-02 committed:
More code with comments on whats left
Mainly whats left to do is display and population logic. I have a couple questions about output format in the comments.

neiroc-02 committed:
Minor corrections

neiroc-02 committed:
Minor correction

neiroc-02 committed:
Added async logic

neiroc-02 committed:
Updated populate keys function

neiroc-02 committed:
Merge pull request #6 from mrElnekave/main

Merge pull request #3 from mrElnekave/mpc-merge

neiroc-02 committed:
Added comments for clarity
